### PR TITLE
chore(refactor): implement Command interface for every command

### PR DESCRIPTION
**Pull request by christiansiegel**

### CODE REVIEW

File: gitopscli/commands/command_factory.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    SyncAppsCommand.Args,
+    VersionCommand.Args,
+]
+
+
+class CommandFactory:
+    @staticmethod
+    def create(args: CommandArgs) -> Command:
+        command: Command
+        if isinstance(args, DeployCommand.Args):
```
Comment (joachimprinzbach) [inline gitopscli/commands/command_factory.py:29]: ich frag mich grad: Könnte man das ned so machen, dass die factory methode jeweils im command ist, also das interface genutzt wird? PseudoCode: 

```python
command: Command 
return command.create(args)
```

Comment (christiansiegel) [reply gitopscli/commands/command_factory.py:29]: You mean having the static `create` in the `Command` interface itself? Wouldn't that be a circular dependency?

Comment (joachimprinzbach) [reply gitopscli/commands/command_factory.py:29]: No not static, that would not work. Let's merge this one, I can think about it later on.

**Review by joachimprinzbach — APPROVED** [review]

**Pull request merged**